src/PSR7/Validators/QueryArgumentsValidator.php

```diff
@@ -75,20 +80,33 @@ private function checkMissingArguments(OperationAddress $addr, array $parsedQuer
     /**
      * @param mixed[]     $parsedQueryArguments [limit=>10]
      * @param Parameter[] $specs
+     *
+     * @throws InvalidQueryArgs
      */
     private function validateAgainstSchema(OperationAddress $addr, array $parsedQueryArguments, int $validationStrategy, array $specs) : void
     {
         // Note: By default, OpenAPI treats all request parameters as optional.
 
         foreach ($parsedQueryArguments as $name => $argumentValue) {
-            // skip if there are no schema for this argument
+            // skip if there is no spec for this argument
             if (! array_key_exists($name, $specs)) {
                 continue;
             }
 
+            $data      = $argumentValue;
+            $parameter = $specs[$name];
+            $schema    = $parameter->schema;
+            if ($schema === null) {
+                // There should be one and only one entry in 'content'.
```

Comment:
It sounds that we rely on some kind of docs here. Can you share a link to it? What happens if we have zero entries? two entries?

Reply:
> It sounds that we rely on some kind of docs here. Can you share a link to it?

I found strict rules on how to use schema/content in [Swagger Editor](https://editor.swagger.io/). By the magic of testing all possible combinations, I found these are considered structural errors:
1. Having both `schema` and `content`
2. Not having `schema` or `content`
3. Having `content` with no properties
4. Having `content` with more than 1 properties

The documentation in Swagger website is a bit vague here, I could only find this relevant quote:
https://swagger.io/docs/specification/describing-parameters/#schema-vs-content
> To describe the parameter contents, you can use either the schema or content keyword. They are mutually exclusive (...)

It looks good enough to me, so I'll add a link to it in the comments., 

> What happens if we have zero entries? two entries?

Both these cases are invalid OpenAPI specifications, however, `cebe/php-openapi` doesn't raise any issues while parsing files with these problems. I also couldn't find in this library (`thephpleague/openapi-psr7-validator`) any exceptions related to errors in the OpenAPI specification itself.
To avoid running into the null type error again, I would suggest one of the following:
- Throw an Exception (ex. InvalidSpecification) that does **not** extend from `ValidationFailed` for any of the 4 invalid combinations I listed above
- Skip parameters that cannot be validated because the specification is invalid.

What do you think it's best here? (I'll implement one of these in this PR to show how it would look like).

Reply:
Yea, I've checked editor too, we should throw invalid schema for count !== 1